ledger/src/blockstore.rs

```diff
@@ -2157,6 +2218,57 @@ impl Blockstore {
                 entry.insert(WorkingEntry::Clean(meta));
             }
         }
+
+        Ok(())
+    }
+
+    fn maybe_update_parent_meta(
+        &self,
+        shred: &Shred,
+        location: BlockLocation,
+        slot: u64,
+        just_inserted_shreds: &mut HashMap<(BlockLocation, ShredId), Cow<'_, Shred>>,
+        parent_meta_working_set: &mut HashMap<(BlockLocation, u64), WorkingEntry<ParentMeta>>,
+    ) -> Result<()> {
+        let key = (location, slot);
+
+        let parent_meta = parent_meta_working_set
```

Comment:
This is not quite right - if we get a blockstore hit we need to update `parent_meta_working_set` (as clean) otherwise we're calling blockstore for every shred that makes it here in the insertion batch.